opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchIndex.java

```diff
@@ -171,19 +171,14 @@ public PhysicalPlan implement(LogicalPlan plan) {
   public TableScanBuilder createScanBuilder() {
     final int querySizeLimit = settings.getSettingValue(Settings.Key.QUERY_SIZE_LIMIT);
 
+    final TimeValue cursorKeepAlive = settings.getSettingValue(Settings.Key.SQL_CURSOR_KEEP_ALIVE);
     var builder = new OpenSearchRequestBuilder(
         querySizeLimit,
         createExprValueFactory());
-
-    return new OpenSearchIndexScanBuilder(builder) {
-      @Override
-      protected TableScanOperator createScan(OpenSearchRequestBuilder requestBuilder) {
-        final TimeValue cursorKeepAlive =
-            settings.getSettingValue(Settings.Key.SQL_CURSOR_KEEP_ALIVE);
-        return new OpenSearchIndexScan(client, requestBuilder.getMaxResponseSize(),
-            requestBuilder.build(indexName, getMaxResultWindow(), cursorKeepAlive));
-      }
-    };
+    Function<OpenSearchRequestBuilder, OpenSearchIndexScan> createScanOperator =
+        requestBuilder -> new OpenSearchIndexScan(client, requestBuilder.getMaxResponseSize(),
+        requestBuilder.build(indexName, getMaxResultWindow(), cursorKeepAlive));
+    return new OpenSearchIndexScanBuilder(builder, createScanOperator);
```

Comment:
If you will do
```suggestion
    return new OpenSearchIndexScanBuilder(new OpenSearchIndexScanQueryBuilder(builder), createScanOperator);
```
You can remove second constructor from `OpenSearchIndexScanBuilder`

Reply:
That was a good idea and I tried it. It requires making `OpenSearchIndexScanQueryBuilder` public which currently stays as an implementation detail. So either extra constructor or extra class dependency.